crates/lair_keystore_api/src/sodium_secretstream.rs

```diff
@@ -108,63 +115,105 @@ where
-        use sodoken::crypto_box::curve25519xchacha20poly1305 as cbox;
-        use sodoken::kx;
-
         // box these up into trait objects so we can easily refer to their types.
         let mut send: PrivRawSend = Box::new(send);
         let mut recv: PrivRawRecv = Box::new(recv);
 
         // read the sealed initiator message
-        let mut cipher: [u8; 64 + cbox::SEALBYTES] = [0; 64 + cbox::SEALBYTES];
+        let mut cipher = [0; 64 + sodoken::crypto_box::XSALSA_SEALBYTES];
         recv.read_exact(&mut cipher).await?;
-        let cipher = sodoken::BufReadSized::from(cipher);
-        let msg = <sodoken::BufWriteSized<64>>::new_no_lock();
-        cbox::seal_open(msg.clone(), cipher, srv_id_pub_key, srv_id_sec_key)
-            .await?;
-        let msg = msg.try_unwrap_sized().unwrap();
-
-        let oth_cbox_pub = sodoken::BufReadSized::from(&msg[..32]);
-        let oth_kx_pub = sodoken::BufReadSized::from(&msg[32..]);
+        let msg = tokio::task::spawn_blocking(move || -> LairResult<_> {
+            let mut msg = [0; 64];
+            sodoken::crypto_box::xsalsa_seal_open(
+                &mut msg,
+                &cipher,
+                &srv_id_pub_key,
+                &srv_id_sec_key.lock().lock(),
+            )?;
+
+            Ok(msg)
+        })
+        .await
+        .map_err(OneErr::new)??;
+
+        let mut oth_cbox_pub: [u8; 32] = [0; 32];
+        oth_cbox_pub.copy_from_slice(&msg[..32]);
+
+        let mut oth_kx_pub: [u8; 32] = [0; 32];
+        oth_kx_pub.copy_from_slice(&msg[32..]);
 
         // generate an ephemeral kx keypair
-        let eph_kx_pub = sodoken::BufWriteSized::new_no_lock();
-        let eph_kx_sec = sodoken::BufWriteSized::new_mem_locked()?;
-        kx::keypair(eph_kx_pub.clone(), eph_kx_sec.clone())?;
+        let (eph_kx_pub, mut eph_kx_sec) =
+            tokio::task::spawn_blocking(move || -> LairResult<_> {
+                let mut eph_kx_pub =
+                    [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+                let mut eph_kx_sec = sodoken::SizedLockedArray::<
+                    { sodoken::crypto_box::XSALSA_SECRETKEYBYTES },
+                >::new()?;
+                sodoken::crypto_box::xsalsa_keypair(
+                    &mut eph_kx_pub,
+                    &mut eph_kx_sec.lock(),
+                )?;
+
+                Ok((eph_kx_pub, eph_kx_sec))
+            })
+            .await
+            .map_err(OneErr::new)??;
 
         // seal our ephemeral kx pub key
-        let cipher =
-            <sodoken::BufWriteSized<{ 32 + cbox::SEALBYTES }>>::new_no_lock();
-        cbox::seal(cipher.clone(), eph_kx_pub.clone(), oth_cbox_pub).await?;
-        let cipher = cipher.try_unwrap_sized().unwrap();
+        let mut cipher =
+            tokio::task::spawn_blocking(move || -> LairResult<_> {
+                let mut cipher = sodoken::SizedLockedArray::<
+                    { 32 + sodoken::crypto_box::XSALSA_SEALBYTES },
+                >::new()?;
```

Comment:
```suggestion
                let mut cipher = [0; 32 + sodoken::crypto_box::XSALSA_SEALBYTES];
```

The cipher is the encrypted keys (actually pub-keys, so not 100% sure why we're encrypting those anyways)... But we don't need to use locked memory for the cipher.

Reply:
https://github.com/holochain/lair/pull/143/commits/b6f64ac499f6c25e33b6b125e4b3e59e90761dca